pkg/server/commands/write_test.go

```
@@ -197,12 +209,227 @@ func TestTransactionalWriteFailedError(t *testing.T) {
+					"owner":  typesystem.This(),
+					"writer": typesystem.This(),
+					"viewer": typesystem.This(),
+				},
+				Metadata: &openfgav1.Metadata{
+					Relations: map[string]*openfgav1.RelationMetadata{
+						"owner": {
+							DirectlyRelatedUserTypes: []*openfgav1.RelationReference{
+								typesystem.DirectRelationReference("user", ""),
+								typesystem.ConditionedRelationReference(
+									typesystem.DirectRelationReference("user", ""),
+									"condition1",
+								),
+							},
+						},
+						"writer": {
+							DirectlyRelatedUserTypes: []*openfgav1.RelationReference{
+								typesystem.DirectRelationReference("user", ""),
+							},
+						},
+						"viewer": {
+							DirectlyRelatedUserTypes: []*openfgav1.RelationReference{
+								typesystem.ConditionedRelationReference(
+									typesystem.WildcardRelationReference("user"),
+									"condition1",
+								),
+							},
+						},
+					},
+				},
+			},
+		},
+		Conditions: map[string]*openfgav1.Condition{
+			"condition1": {
+				Name:       "condition1",
+				Expression: "param1 == 'ok'",
+				Parameters: map[string]*openfgav1.ConditionParamTypeRef{
+					"param1": {
+						TypeName: openfgav1.ConditionParamTypeRef_TYPE_NAME_STRING,
+					},
+				},
+			},
+		},
+	}
+
+	contextStructGood, err := structpb.NewStruct(map[string]interface{}{"param1": "ok"})
+	require.NoError(t, err)
+
+	contextStructBad, err := structpb.NewStruct(map[string]interface{}{"param1": "ok", "param2": 1})
+	require.NoError(t, err)
+
+	logger := logger.NewNoopLogger()
+
+	mockController := gomock.NewController(t)
+	defer mockController.Finish()
+
+	mockDatastore := mockstorage.NewMockOpenFGADatastore(mockController)
+	mockDatastore.EXPECT().MaxTuplesPerWrite().AnyTimes().Return(10)
+	mockDatastore.EXPECT().
+		ReadAuthorizationModel(gomock.Any(), gomock.Any(), gomock.Any()).
+		AnyTimes().
+		Return(model, nil)
+
+	cmd := NewWriteCommand(mockDatastore, logger)
+
+	tests := []test{
```

Would you add a test to cover the case where you reference an undefined condition. For example, `condition2`.

If a tuple references a condition that is undefined in that model, then it should return an error.